exec/java-exec/src/main/java/org/apache/drill/exec/store/dfs/easy/EasySubScan.java

```diff
@@ -47,18 +48,20 @@
 
   @JsonCreator
   public EasySubScan(
-      @JsonProperty("userName") String userName,
-      @JsonProperty("files") List<FileWorkImpl> files,
-      @JsonProperty("storage") StoragePluginConfig storageConfig,
-      @JsonProperty("format") FormatPluginConfig formatConfig,
-      @JacksonInject StoragePluginRegistry engineRegistry,
-      @JsonProperty("columns") List<SchemaPath> columns,
-      @JsonProperty("selectionRoot") Path selectionRoot,
-      @JsonProperty("partitionDepth") int partitionDepth
-      ) throws ExecutionSetupException {
+    @JsonProperty("userName") String userName,
+    @JsonProperty("files") List<FileWorkImpl> files,
+    @JsonProperty("storage") StoragePluginConfig storageConfig,
+    @JsonProperty("format") FormatPluginConfig formatConfig,
+    @JacksonInject StoragePluginRegistry engineRegistry,
+    @JsonProperty("columns") List<SchemaPath> columns,
+    @JsonProperty("selectionRoot") Path selectionRoot,
+    @JsonProperty("partitionDepth") int partitionDepth,
+    @JsonProperty("schema") TupleSchema schema
+    ) throws ExecutionSetupException {
     super(userName);
     this.formatPlugin = (EasyFormatPlugin<?>) engineRegistry.getFormatPlugin(storageConfig, formatConfig);
     Preconditions.checkNotNull(this.formatPlugin);
+    this.formatPlugin.setSchema(schema);
```

Comment:
Again, can't store this in the plugin: the plugin is (or should be) shared.

Or, are we serializing the plugin (config) to handle table properties? If so, then the schema can be set via the Drill web console, which I don't think we want as it will encourage people to create plugin configs for table schema, but that will require that each file have a distinct file suffix, which will cause confusion. (Already does for CSV/CSVH).